iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -225,13 +236,25 @@ PoshRuntime::requestPublisherFromRoudi(const MqMessage& sendBuffer) noexcept
 
 SubscriberPortUserType::MemberType_t*
 PoshRuntime::getMiddlewareSubscriber(const capro::ServiceDescription& service,
-                                     const uint64_t& historyRequest,
+                                     const popo::SubscriberOptions& subscriberOptions,
                                      const NodeName_t& nodeName,
                                      const PortConfigInfo& portConfigInfo) noexcept
 {
+    constexpr uint64_t MAX_QUEUE_CAPACITY = SubscriberPortUserType::MemberType_t::ChunkQueueData_t::MAX_CAPACITY;
+
```

Comment:
We sould also check against MAX_HISTORY. I fear this is missing so far

Reply:
`MAX_HISTORY_CAPACITY` is checked. See a few lines above. What's not checked is `historyRequest`, but I thought since this can be greater than the configured `historyCapacity`, it doesn't matter if it's also greater than `MAX_HISTORY_CAPACITY`

Reply:
OK, that's true. We do not know on subscriber side which historyCapacity is available on publisher side. If the requested history is larger than the available one, we should print a warning on publisher side. Maybe this is even done?

Reply:
do you know where I have to look?

Reply:
I check

Reply:
It is not checked. I remember that I thought about whether to respond with NACK if history request and history capacity do not match. But then we cannot differentiate whether the NACK is because there is no publisher available or because of a history mismatch. And then we need extra info in NACK. Printing a warning in chunk_distributor.inl is all I would do for now. But this can also be done in a follow up PR

Reply:
btw, with the proposed fix for #332 it would be possible to send NACK with error codes. Shall I add a comment for refactoring after #332 is finished?

Reply:
This NACK here is a CaPro message in the discovery loop. What you mention are responses over the message queue. So I'm not sure if you mix something up

Reply:
Whopsi, yes. Although this could also be checked when the SubscriberPort is requested from the runtime

Reply:
Hmm. But only if the publisher port is already present...

Reply:
This is the reason Chuck Norris admires you